components/LoginLogout.tsx

```diff
@@ -26,13 +58,13 @@ export default function LoginLogout() {
 					<li className='border-b-2 border-b-gray-200 p-2 text-center'>
 						<Link href='/u' className='cursor-pointer w-full'>Profile</Link>
 					</li>
-					<li className='border-b-2 border-b-gray-200 p-2 text-center'>
+					<li id='contribute-link' className='border-b-2 border-b-gray-200 p-2 text-center'>
 						<Link href='https://github.com/Alokit-Innovations/' target='_blank' className='cursor-pointer w-full'>Contribute</Link>
 					</li>
-					<li className='border-b-2 border-b-gray-200 p-2 text-center'>
+					<li id='settings-link' className='border-b-2 border-b-gray-200 p-2 text-center'>
 						<Link href='/settings' className='cursor-pointer w-full'>Settings</Link>
 					</li>
-					<li className='p-2 text-center cursor-pointer' onClick={() => logout()}>
+					<li id='logout-link' className='p-2 text-center cursor-pointer' onClick={() => { if (rudderEventMethods) {logout(getAuthUserId(session), getAuthUserName(session), getAndSetAnonymousIdFromLocalStorage(), rudderEventMethods)}}}>
```

Comment:
This restricts logging out if `rudderEventMethods` are not found.
Maybe don't stop a person for being able to log out.